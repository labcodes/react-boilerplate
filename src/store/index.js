import { applyMiddleware, createStore } from 'redux';
import logger from 'redux-logger';

import actions from '../actions';

// ====

const store = createStore(
    actions,
    applyMiddleware(logger)
);

// ====

export default store;
