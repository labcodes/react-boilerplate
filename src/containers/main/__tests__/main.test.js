'use strict';

import React from 'react';

// Test helpers
import renderer from 'react-test-renderer';

// Containers connected
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from 'store';

// Component
import Main from 'containers/main';

// ====

describe('Main [Snapshot]', () => {
    it('Renders correctly', () => {
        const component = (
            <BrowserRouter>
                <Provider store={store}>
                    <Main />
                </Provider>
            </BrowserRouter>
        );

        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
