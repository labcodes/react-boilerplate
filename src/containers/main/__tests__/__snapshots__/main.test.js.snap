// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main [Snapshot] Renders correctly 1`] = `
<main>
  <aside
    className="nav-list"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </aside>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
      <button
        onClick={[Function]}
      >
        change user status
      </button>
    </header>
    <p
      className="App-intro"
    >
      User is unlogged.
    </p>
  </div>
</main>
`;
